idea/meta-storage/src/services/meta/meta.service.ts

```diff
@@ -0,0 +1,51 @@
+import { plainToClass } from 'class-transformer';
+import { HexString } from '@polkadot/util/types';
+import { getProgramMetadata } from '@gear-js/api';
+
+import { CreateMetaInput } from './input/create-meta.input';
+import { Meta } from '../../database/entities';
+import { metaRepository } from '../../database/repositories';
+import { MetaAlreadyExists, MetadataNotFound } from '../../common/errors';
+import { generateHash } from '../../common/helpers/generate-hash';
+
+export const metaService = {
+  async createByIndexer(createMetaInput: CreateMetaInput): Promise<Meta | null> {
```

Comment:
What does `byIndexer` mean? Name `create` would be clear enough. And you can rename `create` function to `update` because it actually does update an existed record

Reply:
And actually it's better to keep just one method

Reply:
https://github.com/gear-tech/gear-js/pull/1282#discussion_r1183954704